extensions/ql-vscode/src/languageSupport.ts

```diff
@@ -0,0 +1,52 @@
+import { IndentAction, languages } from "vscode";
+
+
+/**
+ * OnEnterRules are available in language-configurations, but you cannot specify them in the language-configuration.json.
+ * They can only be specified programmatically.
+ *
+ * Also, we should keep the language-configuration.json as a json file and register it in the package.json because
+ * it is registered first, before the extension is activated, so language features are available quicker.
+ *
+ * See https://github.com/microsoft/vscode/issues/11514
+ * See https://github.com/microsoft/vscode/blob/master/src/vs/editor/test/common/modes/supports/javascriptOnEnterRules.ts
+ */
+export function install() {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const langConfig = require('../language-configuration.json');
+  // setLanguageConfiguration requires a regexp for the wordpattern, not a string
+  langConfig.wordPattern = new RegExp(langConfig.wordPattern);
+  langConfig.onEnterRules = onEnterRules;
+
+  languages.setLanguageConfiguration('ql', langConfig);
+  languages.setLanguageConfiguration('qll', langConfig);
+  languages.setLanguageConfiguration('dbscheme', langConfig);
```

Comment:
Is this right? The dbscheme files have a syntax that's entirely unlike ql and qll

Reply:
In the package.json, dbscheme uses the same language config as ql and qll. So, this is not a regression, though we could consider implementing a different language config in the future.